addons/partner_autocomplete/models/res_partner.py

```diff
@@ -4,14 +4,26 @@
 import base64
 import json
 import logging
+import re
 import requests
 
 from odoo import api, fields, models, tools, _
 
 _logger = logging.getLogger(__name__)
 
 PARTNER_AC_TIMEOUT = 5
-
+# This list is the counterpart to what is used in the Odoo module `directory_service` in the
+# file `autocomplete_company_sync.py`. They don't need to be a perfect match. The client can
+# already anticipatorily sync VAT numbers that aren't supported upstream, but will be. In
+# this case, the upstream `directory_service` will process them, mark them as unsupported,
+# and keep their records in a dormant state until support is added for those VAT formats.
```

Comment:
Avoid mentioning name of files, modules, etc explicitly in comments, it will very probably be outdated at some point and we won't remember to update them.

Also, we shouldn't mention the code of IAP, it's an internal repository to which only a few people have access, but this comment is visible to everyone. Most people who will see this comment won't be able to understand it.